google-cloud-storage/lib/google/cloud/storage/file.rb

```diff
@@ -639,6 +639,9 @@ def rotate encryption_key: nil, new_encryption_key: nil
         # Permanently deletes the file.
         #
         # @return [Boolean] Returns `true` if the file was deleted.
+        # @param [Boolean,Integer] generation Select a specific revision of the
+        #   file to delete. When `true` it will use the object's generation. The
```

Comment:
If the File always uses it's local `generation` value then it is a change in behavior. But that complicates things when we can create a File object that has skipped the lookup, because it won't have the generation loaded. Then a looked up object behaves differently than a skipped looked up object. With all the recent Storage changes being made I think this approach is the right way to go.